userspace/libscap/scap_procs.c

```diff
@@ -642,6 +675,34 @@ static int32_t scap_proc_add_from_proc(scap_t* handle, uint32_t tid, char* procd
 	int32_t res = SCAP_SUCCESS;
 	struct stat dirstat;
 
+
+	if (handle->m_cgroup_version == 0)
+	{
+		snprintf(dir_name, sizeof(dir_name), "%s/filesystems", procdirname);
+		f = fopen(dir_name, "r");
+		if (f)
+		{
+			while(fgets(line, sizeof(line), f) != NULL)
+			{
+				if (strstr(line, "cgroup2"))
+				{
+					handle->m_cgroup_version = 2;
+					break;
+				}
+				if (strstr(line, "cgroup"))
+				{
+					handle->m_cgroup_version = 1;
+				}
+			}
```

Comment:
Are cgroups v2 and v1 exclusive? i.e. can we never have both on a single system? I expect so, but could use a confirmation :)

Reply:
Well in theory they're not; but both podman and docker do not support mixing those ones (see https://github.com/docker/for-linux/issues/1256 for example).
I think we can try to be smarter here, but i think the code would get much more complicated without above assumption.

Reply:
I will add a comment about this, if you're okay with that!

Reply:
Note however that in case of mixed cgroups versions, we will just assume that:
`0::$path` is cgroups v2 unified hierarchy.
All the other cases are same as before.  
Moreover, `0::$path` could also mean, for cgroups v1, 
>  a) the controller is not mounted on a cgroups v1
>                       hierarchy;
> 
>  b) the controller is bound to the cgroups v2 single
>                       unified hierarchy; or
> 
>  c) the controller is disabled (see below).
therefore, at the very worst case, we may end up using an unmounted/disabled controller for cgroups v1 (but how could that happen that a process is bind to an unmounted/disabled controller? I guess it can never happen).

Reply:
@gnosek does it look good for you now? :) 
Thanks!